src/debtManagers/HookAMM.sol

```diff
@@ -0,0 +1,238 @@
+    uint public feesAccrued;
+
+    error Invariant();
+
+    /**
+     * @dev Constructor for sDola contract.
+     * WARNING: MIN_SHARES will always be unwithdrawable from the vault. Deployer should deposit enough to mint MIN_SHARES to avoid causing user grief.
+     * @param _dola Address of the DOLA token.
+     * @param _gov Address of the governance.
+     * @param _K Initial value for the K variable used in calculations.
+     */
+    constructor(
+        address _dola,
+        address _dbr,
+        address _gov,
+        address _feeRecipient,
+        uint _K
+    ) {
+        require(_K > 0, "_K must be positive");
+        dbr = Mintable(_dbr);
+        dola = IERC20(_dola);
+        gov = _gov;
+        feeRecipient = _feeRecipient;
+        targetK = _K;
+    }
+
+    modifier onlyGov() {
+        require(msg.sender == gov, "ONLY GOV");
+        _;
+    }
+
+    modifier buyHook() {
+        variableDebtManager.buyHook();
+        _;
+    }
+
+    /**
+     * @dev Returns the current value of K, which is a weighted average between prevK and targetK.
+     * @return The current value of K.
+     */
+    function getK() public view returns (uint) {
+        uint duration = 7 days;
+        uint timeElapsed = block.timestamp - lastKUpdate;
+        if(timeElapsed > duration) {
+            return targetK;
+        }
+        uint targetWeight = timeElapsed;
+        uint prevWeight = duration - timeElapsed;
+        return (prevK * prevWeight + targetK * targetWeight) / duration;
+    }
+
+    /**
+     * @dev Calculates the DOLA reserve based on the current DBR reserve.
+     * @return The calculated DOLA reserve.
+     */
+    function getDolaReserve() public view returns (uint) {
+        return dola.balanceOf(address(this)) + getVirtualDolaReserves();
+    }
+
+    /**
+     * @dev Returns the current DBR reserve as the sum of dbr balance and claimable dbr
+     * @return The current DBR reserve.
+     */
+    function getDbrReserve() public view returns (uint) {
+        return getK() / getDolaReserve();
+    }
+
+    /**
+     * @dev Sets a new target K value.
+     * @param _K The new target K value.
+     */
+    function setTargetK(uint _K) external onlyGov {
+        require(_K > getDbrReserve(), "K must be larger than dbr reserve");
+        prevK = getK();
+        targetK = _K;
+        lastKUpdate = block.timestamp;
+        emit SetTargetK(_K);
+    }
+
+    function setMaxDolaPrice(uint _maxDolaPrice) external onlyGov {
+        require(_maxDolaPrice >= minDolaPrice, "Max DOLA price must be higher or equal to min DOLA price");
+        maxDolaPrice = _maxDolaPrice;
+    }
+
+    function setMinDolaPrice(uint _minDolaPrice) external onlyGov {
+        require(maxDolaPrice >= _minDolaPrice, "Max DOLA price must be higher or equal to min DOLA price");
+        minDolaPrice = _minDolaPrice;
+    }
+
+    function getVirtualDolaReserves() internal view returns(uint) {
+        return sqrt(1e18 * getK() / minDolaPrice);
```

Comment:
Can you explain what's going on here? Why is minDolaPrice used here? and how does this square root result in dola reserves? Shouldn't we be dividing K by DBR reserve to get DOLA reserves?